d2l/torch.py

```diff
@@ -3071,6 +3071,136 @@ def predict_seq2seq(net, src_sentence, src_vocab, tgt_vocab, num_steps,
         output_seq.append(pred)
     return ' '.join(tgt_vocab.to_tokens(output_seq)), attention_weight_seq
 
+def FrozenLake(seed):
+    """Defined in :numref:`sec_code_rl`"""
+    '''
+            Winter is here. You and your friends were tossing around a frisbee at the
+            park when you made a wild throw that left the frisbee out in the middle of
+            the lake. The water is mostly frozen, but there are a few holes where the
+            ice has melted. If you step into one of those holes, you'll fall into the
+            freezing water. At this time, there's an international frisbee shortage, so
+            it's absolutely imperative that you navigate across the lake and retrieve
+            the disc. However, the ice is slippery, so you won't always move in the
+            direction you intend.
+            The surface is described using a grid like the following
+                SFFF
+                FHFH
+                FFFH
+                HFFG
+            S : starting point, safe
+            F : frozen surface, safe
+            H : hole, fall to your doom
+            G : goal, where the frisbee is located
+            The episode ends when you reach the goal or fall in a hole.
+            You receive a reward of 1 if you reach the goal, and zero otherwise.
+            You can only move in the following directions:
+                LEFT which corresponds to action index 0.
+                DOWN which corresponds to action index 1.
+                RIGHT which corresponds to action index 2.
+                UP which corresponds to action index 3.
+            Above text/description is copied from https://gym.openai.com/envs/FrozenLake-v0/
+            and how to process env.P.items is adpated from https://sites.google.com/view/deep-rl-bootcamp/labs
+
+    '''
+    import gym
+    env = gym.make('FrozenLake-v1', is_slippery=False)
+    env.seed(seed)
+    env.action_space.np_random.seed(seed)
+    env.action_space.seed(seed)
+    env_info = {}
+    env_info['desc'] = env.desc  # 2D array specifying what each grid item means
+    env_info['num_states']     = env.nS # number of observations/states or obs/state dim
+    env_info['num_actions']    = env.nA # number of actions or action dim
+    # define indices for (transition probability, nextstate, reward, done) tuple
+    env_info['trans_prob_idx'] = 0 # index of transition probability entry
+    env_info['nextstate_idx']  = 1 # index of next state entry
+    env_info['reward_idx']     = 2 # index of reward entry
+    env_info['done_idx']       = 3 # index of done entry
```

Comment:
Please apply consistent code style here: 2 spaces before # for inline comments; Capitalize the first letters of code comments; one space around operators.

Reply:
Done.